src/Menu/CachedFrontendMenuBuilder.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bolt\Menu;
+
+use Bolt\Collection\DeepCollection;
+use Psr\SimpleCache\CacheInterface;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+class CachedFrontendMenuBuilder
+{
+    /** @var CacheInterface */
+    private $cache;
+
+    /** @var FrontendMenuBuilder */
+    private $menuBuilder;
+
+    /** @var Stopwatch */
+    private $stopwatch;
+
+    public function __construct(CacheInterface $cache, FrontendMenuBuilder $menuBuilder, Stopwatch $stopwatch)
```

Comment:
remove stopwatch from here or move it to another decorator

Reply:
No, let's not get overboard with the abstractions.

Reply:
But then stopwatch will have to be included as dependency in prod env